internal/oci/notation/notation.go

```diff
@@ -0,0 +1,388 @@
+/*
+Copyright 2023 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notation
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	_ "github.com/notaryproject/notation-core-go/signature/cose"
+	_ "github.com/notaryproject/notation-core-go/signature/jws"
+	"github.com/notaryproject/notation-go"
+	"github.com/notaryproject/notation-go/registry"
+	verifier "github.com/notaryproject/notation-go/verifier"
+	"github.com/notaryproject/notation-go/verifier/trustpolicy"
+	"github.com/notaryproject/notation-go/verifier/truststore"
+	oras "oras.land/oras-go/v2/registry/remote"
+	oauth "oras.land/oras-go/v2/registry/remote/auth"
+	retryhttp "oras.land/oras-go/v2/registry/remote/retry"
+
+	"github.com/fluxcd/source-controller/internal/helm/common"
+	"github.com/fluxcd/source-controller/internal/oci"
+)
+
+// name of the trustpolicy file defined in the Secret containing
+// notation public keys.
+const DefaultTrustPolicyKey = "trustpolicy.json"
+
+// options is a struct that holds options for verifier.
+type options struct {
+	rootCertificate [][]byte
```

Comment:
```suggestion
	rootCertificates [][]byte
```

Reply:
Resolved in [271aaa4](https://github.com/fluxcd/source-controller/pull/1075/commits/271aaa4961eed43aef8ca7d9f54f252047fcd340).